runtime/oti/j9.h

```diff
@@ -323,8 +323,12 @@ static const struct { \
 /* Macros for ValueTypes */
 #ifdef J9VM_OPT_VALHALLA_VALUE_TYPES
 #define J9_IS_J9CLASS_VALUETYPE(clazz) J9_ARE_ALL_BITS_SET(clazz->classFlags, J9ClassIsValueType)
+#define J9_IS_OBJECT_OR_VALUETYPE(sigChar) (('L' == sigChar) || ('Q' == sigChar))
```

Comment:
```c
#define J9_IS_OBJECT_OR_VALUETYPE(sigChar) (('L' == (sigChar)) || ('Q' == (sigChar)))
```

Reply:
Always add extra brackets around the value passed into the macro to ensure that it's evaluated as expected.

Without the brackets, something like
`J9_IS_OBJECT_OR_VALUETYPE(a || b)` would evaluate as `('L' != a) || b`, which while a stupid example, illustrates the potential issue.